libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt

```diff
@@ -130,128 +159,120 @@ class RouteLineColorResources private constructor(
+            return false
+        if (alternativeRouteSevereCongestionColor != other.alternativeRouteSevereCongestionColor)
+            return false
+        if (alternativeRouteUnknownCongestionColor != other.alternativeRouteUnknownCongestionColor)
             return false
-        if (restrictedRoadColor != other.restrictedRoadColor) return false
         if (alternativeRouteRestrictedRoadColor != other.alternativeRouteRestrictedRoadColor)
             return false
-        if (routeClosureColor != other.routeClosureColor) return false
         if (alternativeRouteClosureColor != other.alternativeRouteClosureColor) return false
-        if (inActiveRouteLegsColor != other.inActiveRouteLegsColor)
-            return false
+        if (routeLineTraveledColor != other.routeLineTraveledColor) return false
+        if (routeLineTraveledCasingColor != other.routeLineTraveledCasingColor) return false
+        if (routeCasingColor != other.routeCasingColor) return false
+        if (alternativeRouteCasingColor != other.alternativeRouteCasingColor) return false
+        if (inActiveRouteLegsColor != other.inActiveRouteLegsColor) return false
 
         return true
     }
 
-    /**
-     * Returns a hash code value for the object.
-     */
-    override fun hashCode(): Int {
-        var result = routeLineTraveledColor
-        result = 31 * result + routeLineTraveledCasingColor
-        result = 31 * result + routeUnknownTrafficColor
-        result = 31 * result + routeDefaultColor
-        result = 31 * result + routeLowCongestionColor
-        result = 31 * result + routeModerateColor
-        result = 31 * result + routeHeavyColor
-        result = 31 * result + routeSevereColor
-        result = 31 * result + routeCasingColor
-        result = 31 * result + alternativeRouteUnknownTrafficColor
-        result = 31 * result + alternativeRouteDefaultColor
-        result = 31 * result + alternativeRouteLowColor
-        result = 31 * result + alternativeRouteModerateColor
-        result = 31 * result + alternativeRouteHeavyColor
-        result = 31 * result + alternativeRouteSevereColor
-        result = 31 * result + alternativeRouteCasingColor
-        result = 31 * result + restrictedRoadColor
-        result = 31 * result + alternativeRouteRestrictedRoadColor
-        result = 31 * result + routeClosureColor
-        result = 31 * result + alternativeRouteClosureColor
-        result = 31 * result + inActiveRouteLegsColor
-        return result
-    }
-
     /**
      * A builder for instantiating the RouteLineResources class
      */
     class Builder {
-        private var routeLineTraveledColor: Int = RouteConstants.ROUTE_LINE_TRAVELED_COLOR
-        private var routeUnknownTrafficColor: Int = RouteConstants.ROUTE_UNKNOWN_TRAFFIC_COLOR
+        private var lowCongestionRange: IntRange = 1..39 // Should we make it 0..39 since 0 is very very low
+        private var moderateCongestionRange: IntRange = lowCongestionRange.last + 1..59
```

Comment:
Why does this overlap with `lowCongestionRange`?  A value of 35 would be both low and moderate. That doesn't appear correct.

Reply:
How do you mean? 
`moderateCongestionRange = lowCongestionRange.last + 1..59`
i.e.
`moderateCongestionRange = 40..59`
as `lowCongestionRange.last = 39`

Reply:
The above is correct but instead of doing `.last+1` I have removed it and rather assigned concrete values. If not then if a user does `RouteLineColorResources.Builder().lowCongestionRange(0..24).build()`, it will change the `lowCongestionRange` but the moderate would still be `40..59` based on `lowCongestionRange` from the default implementation. 
Hence, we would have to ask the developers to make sure that they cover all the values from 0..100. If they miss any value, it would be treated as default color.